src/test/java/StudentSetTest.java

```diff
@@ -0,0 +1,225 @@
+import com.endava.internship.collections.Student;
+import com.endava.internship.collections.StudentSet;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
```

Comment:
Static imports should be right after the package name, so consider moving them above. 
And also please remove wildcard import, use only needed imports